app/src/main/java/org/p2p/wallet/common/adapter/CommonAnyCellAdapter.kt

```diff
@@ -9,7 +9,30 @@ import org.p2p.uikit.model.AnyCellItem
 class CommonAnyCellAdapter(
     vararg delegates: AdapterDelegate<List<AnyCellItem>>,
     diffUtilCallback: DiffUtil.ItemCallback<AnyCellItem> = DefaultDiffCallback()
-) : AsyncListDifferDelegationAdapter<AnyCellItem>(diffUtilCallback, *delegates)
+) : AsyncListDifferDelegationAdapter<AnyCellItem>(diffUtilCallback, *delegates) {
+
+    @Suppress("UNCHECKED_CAST")
+    fun <T : AnyCellItem> updateItem(
+        predicate: (AnyCellItem) -> Boolean,
+        transform: (T) -> T,
+        animateChanges: Boolean = false
+    ) {
+        // this.items is an ImmutableList so we can't just set the single element
+        val oldItems = items.toMutableList()
+        val index = oldItems.indexOfFirst(predicate)
+        if (index != -1) {
+            val old = oldItems[index] as T
+            val new = transform(old)
+            items = oldItems.apply { set(index, new) }
+            if (animateChanges) {
+                notifyItemChanged(index)
+            } else {
+                // workaround for disabling animation
+                notifyItemChanged(index, Unit)
+            }
```

Comment:
looks complex (

Reply:
it is, but this way we have ability to change only specific cell without remapping the entire state